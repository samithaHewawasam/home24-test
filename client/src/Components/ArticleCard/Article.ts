export const article = {
  name: 'Beistelltisch Hoya (2er-Set)',
  index: 1,
  prices: {
    currency: 'EUR',
    regular: {
      value: 11499
    }
  },
  images: [
    {
      path: 'https://cdn1.home24.net/images/media/catalog/product/200x200/png/-/1/-1000174115-190524-15511900114-IMAGE-P000000001000174115.webp'
    }
  ]
}
