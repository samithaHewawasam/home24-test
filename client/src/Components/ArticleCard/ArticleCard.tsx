import { ShoppingCartOutlined } from '@ant-design/icons'
import { Card, Col, Image, Statistic } from 'antd'
import { Article } from 'Interfaces'
import React, { memo } from 'react'
import { formatter } from 'Utility'

interface IArticleCard {
  article: Article
  handleCartUpdate: any
}

const ArticleCard: React.FunctionComponent<IArticleCard> = memo(function ({
  article,
  handleCartUpdate
}: IArticleCard) {
  return (
    <Col
      xxl={4}
      xl={5}
      lg={6}
      md={6}
      sm={24}
      xs={24}
      style={{ textAlign: 'center' }}
    >
      <Card
        title={article.name}
        hoverable
        actions={[
          <ShoppingCartOutlined
            key={'ShopingCart'}
            onClick={() => {
              handleCartUpdate(article)
            }}
          />
        ]}
      >
        <Image
          alt={article.name}
          data-testid='article-image'
          src={article.images[0].path}
          style={{ marginTop: 5 }}
        />
        <Statistic
          value={formatter.format(article.prices.regular.value / 100)}
          precision={2}
        />
      </Card>
    </Col>
  )
})

ArticleCard.displayName = 'ArticleCard'
export default ArticleCard
