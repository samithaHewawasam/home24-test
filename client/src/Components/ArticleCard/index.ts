export { default as ArticleCard } from './ArticleCard'
export * from './Article'
