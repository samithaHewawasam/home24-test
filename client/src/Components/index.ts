import { Header } from './Header'
import { SideBar } from './SideBar'
import { MainContent } from './MainContent'
import { SearchContent } from './SearchContent'

export { Header, SideBar, MainContent, SearchContent }
