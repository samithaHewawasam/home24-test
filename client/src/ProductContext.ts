import { createContext } from 'react'
import { intialData } from 'ProductIntialData'

/**
 * Conext for Products
 * @param  {State<{ categories: Category[] }>} intialData
 */

export const ProductContext = createContext(intialData)
