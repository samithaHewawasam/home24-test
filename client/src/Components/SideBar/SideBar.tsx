import { MenuOutlined } from '@ant-design/icons'
import { Affix, Layout, Menu } from 'antd'
import { ProductContext } from 'ProductContext'
import React, { memo, useContext } from 'react'
import { useNavigate } from 'react-router-dom'

const { Sider } = Layout

const SideBar: React.FC = memo(() => {
  const { data } = useContext(ProductContext)
  const navigate = useNavigate()

  if (data === null) {
    return <>Error loading Side Bar</>
  }
  const { categories } = data

  return (
    <Affix>
      <Sider
        theme="light"
        style={{
          overflowY: 'auto',
          maxHeight: '100vh',
          overflowX: 'hidden'
        }}
      >
        {categories !== null && (
          <Menu
            mode="inline"
            title=""
            defaultSelectedKeys={['1']}
            defaultOpenKeys={['sub1']}
            style={{ height: '100%' }}
            onSelect={({ key }) => {
              navigate(`/search?category=${key}`)
            }}
            items={[
              {
                key: 'menu',
                label: 'Kategorien',
                icon: <MenuOutlined />
              },
              ...categories[0]?.childrenCategories?.list?.map(
                ({ name }: any, index: number) => ({
                  key: String(index + 1),
                  label: name
                })
              )
            ]}
          />
        )}
      </Sider>
    </Affix>
  )
})

SideBar.displayName = 'SideBar'
export default SideBar
