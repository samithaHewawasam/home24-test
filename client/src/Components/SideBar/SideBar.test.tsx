import { cleanup, render } from '@testing-library/react'
import { act } from '@testing-library/react-hooks/dom'
import { article } from 'Components/ArticleCard'
import { ProductContext } from 'ProductContext'
import React from 'react'
import { BrowserRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'
import SideBar from './SideBar'

const contextData = {
  data: {
    categories: [
      {
        name: '',
        articleCount: 0,
        childrenCategories: {
          list: [
            {
              name: 'Schlafzimmer',
              urlPath: ''
            }
          ]
        },
        categoryArticles: { articles: [article] }
      }
    ]
  },
  isError: false,
  isLoading: false
}

describe('Main Content', () => {
  const SideBarComponent = (
    <ProductContext.Provider value={contextData}>
      <BrowserRouter>
        <SideBar />
      </BrowserRouter>
    </ProductContext.Provider>
  )
  beforeEach(async () => {
    await act(async () => {
      render(SideBarComponent)
    })
  })

  afterEach(() => {
    cleanup()
  })

  /**
   * @description SnapShot for SideBar
   */
  it('renders SideBar correctly', () => {
    const tree = renderer.create(SideBarComponent).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('render the 1st menu as Kategorien', () => {
    const _MenuItems = document.querySelectorAll('.ant-menu-title-content')
    expect(_MenuItems[0].innerHTML).toContain('Kategorien')
  })

  it('render the 2nd menu as Schlafzimmer', () => {
    const _MenuItems = document.querySelectorAll('.ant-menu-title-content')
    expect(_MenuItems[1].innerHTML).toContain('Schlafzimmer')
  })
})
