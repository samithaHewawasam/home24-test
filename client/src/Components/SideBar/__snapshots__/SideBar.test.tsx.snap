// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Content renders SideBar correctly 1`] = `
<div>
  <div
    className=""
  >
    <aside
      className="ant-layout-sider ant-layout-sider-light"
      style={
        Object {
          "flex": "0 0 200px",
          "maxHeight": "100vh",
          "maxWidth": "200px",
          "minWidth": "200px",
          "overflowX": "hidden",
          "overflowY": "auto",
          "width": "200px",
        }
      }
    >
      <div
        className="ant-layout-sider-children"
      >
        <ul
          className="ant-menu ant-menu-root ant-menu-inline ant-menu-light"
          data-menu-list={true}
          onKeyDown={[Function]}
          role="menu"
          style={
            Object {
              "height": "100%",
            }
          }
          tabIndex={0}
          title=""
        >
          <li
            className="ant-menu-item"
            data-menu-id={null}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            role="menuitem"
            style={
              Object {
                "paddingLeft": 24,
              }
            }
            tabIndex={-1}
          >
            <span
              aria-label="menu"
              className="anticon anticon-menu ant-menu-item-icon"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="menu"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M904 160H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0 624H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0-312H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8z"
                />
              </svg>
            </span>
            <span
              className="ant-menu-title-content"
            >
              Kategorien
            </span>
          </li>
          <li
            className="ant-menu-item ant-menu-item-selected ant-menu-item-only-child"
            data-menu-id={null}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            role="menuitem"
            style={
              Object {
                "paddingLeft": 24,
              }
            }
            tabIndex={-1}
          >
            <span
              className="ant-menu-title-content"
            >
              Schlafzimmer
            </span>
          </li>
        </ul>
        <div
          aria-hidden={true}
          style={
            Object {
              "display": "none",
            }
          }
        />
      </div>
    </aside>
  </div>
</div>
`;
